client/trino-client/src/test/java/io/trino/client/auth/external/TestExternalAuthenticator.java

```diff
@@ -119,13 +129,95 @@ public void testAuthentication()
                 .isEqualTo("Bearer valid-token");
     }
 
+    @Test
+    public void testAuthenticationFromMultipleThreadsWithCashedToken()
```

Comment:
Can you please also add a tests with multiple threads and no caching, just in case.